core/src/test/scala/com/softwaremill/events/EventMachineTest.scala

```diff
@@ -90,6 +94,25 @@ class EventMachineTest extends FlatSpec with Matchers with SqlSpec with Eventual
       m.asyncEventRunner.stop()
     }
   }
+
+  it should "recover stored events" in {
+    // given
+    var actions = Vector.empty[String]
+
+    val m = createModules(Registry()
+      .registerModelUpdate[Event1] { e => DBIOAction.successful(()).map { r => actions :+= "mu1" + e.data.data; r } }
+      .registerEventListener[Event1] { e => DBIOAction.successful(Nil).map { r => actions :+= "el1"; r } })
+
+    implicit val hc = HandleContext.System
+    val result = m.eventMachine.run(CommandResult.successful((), Event(Event1("x")).forNewAggregate)).futureValue
+
+    // when
+    val failOverFuture = m.eventMachine.failOverFromStoredEventsTo().futureValue
```

Comment:
Actually, one more thought: maybe it would make more sense to fail over using events between a range of IDs? The ids are time-based so it's quite easy to generate an ID corresponding to a given timestamp (maybe we'd need to extend the id-generator for that).

Two use cases here:

* restore the system as it was at a given time
* having a model with all events up to X applied, apply the new events (after X)

Also, id is the primary key so automatically indexed, that will behave better then a query on the date